pymeasure/instruments/thorlabs/elliptec_utils/base.py

```diff
@@ -0,0 +1,209 @@
+
+        Attributes are needed for physical units from steps (for Rotator or Linear Channel for instance)
+        """
+        info = self.idn
+        if info is None:
+            raise ExternalDeviceNotFound
+        else:
+            self.info = info
+
+            self.range = self.info['Range']
+            self.pulse_per_rev = self.info['Pulse/Rev']
+            self.serial_no = self.info['Serial No.']
+            self.motor_type = self.info['Motor Type']
+
+    def home(self, direction: str = 'forward') -> None:
+        """ Homing forward or backward
+
+        :param direction:
+        """
+        if direction == 'forward':
+            self.ask('ho0')
+        else:
+            self.ask('ho1')
+
+    def move(self, position: int, move_type: str = 'abs') -> None:
+        """ Move the MotorChannel to an absolute or relative position in steps
+
+        :param position: value to move in steps
+        :param move_type: either 'abs' for an absolute move or something else for relative move
+        """
+        if move_type == 'abs':
+            command = 'ma'
+        else:
+            command = 'mr'
+        command += self.int_to_hex(position)
+        self.ask(command)
+
+    @staticmethod
+    def int_to_hex(number: int):
+        """ Convert integer to uppercase hexadecimal character """
+        return int_to_hex(number)
+
+
+class ElliptecController(Instrument):
+    """ An Elliptec Controller adding motor channels at given addresses
+
+    :arg channels: dictionary mapping from an integer as a string ('0', '1', ...) to a MotorChannel or inherited
+    objects
+    """
+
+    def __init__(self, adapter, name="Elliptec Rotator", klass=MotorChannel, address: Union[int, List] = 0, **kwargs):
+        super().__init__(
+            adapter,
+            name,
+            **kwargs)
+
+        if address is not None:
+            self.add_motor_channel(klass, address)
+
+    def add_motor_channel(self, klass=MotorChannel, address: Union[int, List] = 0):
+        """ Add one or several MotorChannel instance (or subclassed) with a given address to the Motor channels
+
+        :param klass: A MotorChannel object or any inherited Motor (LinearChannel, RotatorChannel...)
+        :param address: integer or list of integer as Motor address
+        """
+        if not isinstance(address, Sequence):
+            address = [address]
+
+        for ad in address:
+            self.add_child(klass, str(ad))
+            try:
+                self.load_motor_info()
+            except VisaIOError:
+                raise ExternalDeviceNotFound
+
+    def load_motor_info(self, address: int = 0):
+        """ Asks motor for info and load response into properties """
+        self.channels[str(address)].load_motor_info()
+
+    def close(self):
+        """ Close the adapter communication, here the COM port"""
+        try:
+            self.adapter.close()
+        except:
```

Comment:
Like the linter says: You should state the Exceptions you expect to happen and pass only, if it is an expected use case.